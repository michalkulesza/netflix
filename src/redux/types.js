//error.js
export const SET_ERROR = "SET_ERROR";
export const CLEAR_ERROR = "CLEAR_ERROR";

//fetch-details.js
export const FETCH_DETAILS = "FETCH_DETAILS";
export const CLEAR_DETAILS = "CLEAR_DETAILS";

//fetch-episodes.js
export const FETCH_EPISODES = "FETCH_EPISODES";
export const SET_EPISODES_UPDATING = "SET_EPISODES_UPDATING";
export const CLEAR_EPISODES = "CLEAR_EPISODES";

//fetch-genre-data.js
export const FETCH_GENRE_DATA = "FETCH_GENRE_DATA";
export const FETCH_MORE_GENRE_DATA = "FETCH_MORE_GENRE_DATA";
export const SET_GENRE_DATA_UPDATING = "SET_GENRE_DATA_UPDATING";
export const CLEAR_GENRE_DATA = "CLEAR_GENRE_DATA";

//fetch-initial-data.js
export const SET_INITIAL_DATA = "SET_INITIAL_DATA";
export const SET_DATA_UPDATING = "SET_DATA_UPDATING";
export const CLEAR_INITIAL_DATA = "CLEAR_INITIAL_DATA";

//genres.js
export const SET_GENRES = "SET_GENRES";
export const CLEAR_GENRES = "CLEAR_GENRES";
export const SET_SELECTED_GENRE = "SET_SELECTED_GENRE";
export const CLEAR_SELECTED_GENRE = "CLEAR_SELECTED_GENRE";

//misc.js
export const SET_DETAILS_POSITION = "SET_DETAILS_POSITION";
export const SET_EXPANDED_POSITION = "SET_EXPANDED_POSITION";
export const CLEAR_EXPANDED_POSITION = "CLEAR_EXPANDED_POSITION";
export const SET_EXPANDED_TRANSFORM_ORIGIN = "SET_EXPANDED_TRANSFORM_ORIGIN";
export const SET_HEADER_VIDEO = "SET_HEADER_VIDEO";
export const SET_PRESSED_KEY = "SET_PRESSED_KEY";
export const SET_SCROLLBAR_WIDTH = "SET_SCROLLBAR_WIDTH";
export const SET_ACTIVE_EXPANDED = "SET_ACTIVE_EXPANDED";
export const CLEAR_ACTIVE_EXPANDED = "CLEAR_ACTIVE_EXPANDED";
export const CLEAR_MISC = "CLEAR_MISC";

//player.js
export const SET_PLAYER_TV = "SET_PLAYER_TV";
export const SET_PLAYER_FILM = "SET_PLAYER_FILM";
export const SET_PLAYER_VOLUME = "SET_PLAYER_VOLUME";
export const SET_PLAYER_SEASONS = "SET_PLAYER_SEASONS";
export const SET_PLAYER_STATE = "SET_PLAYER_STATE";
export const SET_PLAYER_META_LOADED = "SET_PLAYER_META_LOADED";
export const CLEAR_PLAYER = "CLEAR_PLAYER";

//toggles.js
export const SET_IS_DETAILS = "SET_IS_DETAILS";
export const SET_GLOBAL_MUTE = "SET_GLOBAL_MUTE";
export const CLEAR_TOGGLES = "CLEAR_TOGGLES";

//user.js
export const SET_USER_INFO = "SET_USER_INFO";
export const SET_USER = "SET_USER";
export const TOGGLE_LIKE_VIDEO = "TOGGLE_LIKE_VIDEO";
export const DISLIKE_VIDEO = "DISLIKE_VIDEO";
export const REMOVE_DISLIKED_VIDEO = "REMOVE_DISLIKED_VIDEO";
export const TOGGLE_VIDEO_LIST = "TOGGLE_VIDEO_LIST";
export const CLEAR_USER = "CLEAR_USER";
