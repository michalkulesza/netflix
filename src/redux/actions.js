import {
	FETCH_INITIAL_DATA,
	FETCH_DETAILS,
	CLEAR_DETAILS,
	FETCH_EPISODES,
	SET_DETAILS,
	SET_DETAILS_POSITION,
	SET_SCROLLED,
} from "./types";
import axios from "axios";

export const fetchInitialData = () => {
	return async dispatch => {
		try {
			const response = await axios.get("http://localhost:8888/initial");

			dispatch({
				type: FETCH_INITIAL_DATA,
				payload: response.data,
			});
		} catch (error) {
			console.error(error.message);
		}
	};
};

export const fetchDetailsMovie = id => {
	return async dispatch => {
		try {
			const response = await axios.post("http://localhost:8888/details/movie", { id });

			dispatch({
				type: FETCH_DETAILS,
				payload: response.data,
			});
		} catch (error) {
			console.error(error.message);
		}
	};
};

export const fetchDetailsTv = id => {
	return async dispatch => {
		try {
			const response = await axios.post("http://localhost:8888/details/tv", { id });
			const responseEpisodes = await axios.post("http://localhost:8888/episodes", { id, season: 1 });

			dispatch({
				type: FETCH_DETAILS,
				payload: response.data,
			});
			dispatch({
				type: FETCH_EPISODES,
				payload: responseEpisodes.data,
			});
		} catch (error) {
			console.error(error.message);
		}
	};
};

export const clearDetails = () => {
	return {
		type: CLEAR_DETAILS,
	};
};

export const fetchEpisodes = (id, season) => {
	return async dispatch => {
		try {
			const response = await axios.post("http://localhost:8888/episodes", { id, season });
			dispatch({
				type: FETCH_EPISODES,
				payload: response.data,
			});
		} catch (error) {
			console.error(error.message);
		}
	};
};

export const setDetails = bool => {
	return {
		type: SET_DETAILS,
		payload: { isDetails: bool },
	};
};

export const setDetailsPosition = (x, y, width, height) => {
	return {
		type: SET_DETAILS_POSITION,
		payload: { detailsPosition: { x, y, width, height } },
	};
};

export const setScrolled = scrolled => {
	return {
		type: SET_SCROLLED,
		payload: { scrolled },
	};
};
