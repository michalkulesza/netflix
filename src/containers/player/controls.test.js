import React from "react";
import { mount } from "enzyme";
import ControlsContainer from "./controls";
import * as types from "../../redux/types";
import configureStore from "redux-mock-store";
import * as ReactRedux from "react-redux";

let useDispatch;
let store = configureStore()({});

beforeEach(() => {
	useDispatch = jest.spyOn(ReactRedux, "useDispatch");
	store.clearActions();
});

describe("<ControlsContainer/> with no props & state", () => {
	const wrapper = mount(
		<ReactRedux.Provider store={store}>
			<ControlsContainer />
		</ReactRedux.Provider>
	);

	it("should match the snapshot", () => {
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("should return null", () => {
		expect(wrapper.isEmptyRender()).toEqual(true);
	});
});

describe("<ControlsContainer/> props & state", () => {
	store = configureStore()({
		player: {
			type: "movie",
			title: "Testtitle",
			src: "/",
			backdrop: "/",
			description: "Lorem ipsum dolor sit amet consectetur adipisicing elit. Itaque, sunt?",
			ep_title: null,
			ep_number: null,
			ep_season: null,
			seasons: null,
			year: 2020,
			ageRestriction: 18,
			runtime: 199,
			volume: 0.6,
			state: "paused",
			metaLoaded: false,
		},
	});
	const handleClickPlay = jest.fn();
	const playerRef = {
		current: {
			currentTime: 0,
			duration: 198,
		},
	};

	const wrapper = mount(
		<ReactRedux.Provider store={store}>
			<ControlsContainer handleClickPlay={handleClickPlay} playerRef={playerRef} />
		</ReactRedux.Provider>
	);

	it("should match the snapshot", () => {
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("should not return null", () => {
		expect(wrapper.isEmptyRender()).toEqual(false);
	});
});
