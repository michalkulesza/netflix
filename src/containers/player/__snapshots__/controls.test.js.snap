// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ControlsContainer/> props & state should match the snapshot 1`] = `"<div class=\\"controls__Main-sc-1r0nwd9-0 eCdxwt\\"><div class=\\"seek-bar__Container-sc-33ruba-0 ecXbym\\"><div class=\\"seek-bar__Main-sc-33ruba-3 jIuMwp\\"><div class=\\"seek-bar__SeekBarContainer-sc-33ruba-5 dGZLvR\\"><div class=\\"seek-bar__SeekBarTotal-sc-33ruba-2 jvKjIW\\"></div></div><div class=\\"seek-bar__SeekIndicator-sc-33ruba-1 dKhQVr\\"></div></div><div class=\\"seek-bar__Time-sc-33ruba-4 cmA-DKU\\">0:03:18</div></div><div class=\\"controls__ButtonsContainer-sc-1r0nwd9-1 iFLaWV\\"><div class=\\"controls__Left-sc-1r0nwd9-2 ePrTkr\\"><button class=\\"button__Clear-nas94j-5 hTVqGS\\"><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 24 24\\" height=\\"1em\\" width=\\"1em\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path fill=\\"none\\" stroke=\\"#000\\" stroke-width=\\"2\\" d=\\"M3,21 L9,21 L9,3 L3,3 L3,21 Z M4,19 L8,19 L8,5 L4,5 L4,19 Z M5,17 L7,17 L7,7 L5,7 L5,17 Z M15,21 L21,21 L21,3 L15,3 L15,21 Z M16,19 L20,19 L20,5 L16,5 L16,19 Z M17,17 L19,17 L19,7 L17,7 L17,17 Z\\"></path></svg></button><button class=\\"button__Clear-nas94j-5 hTVqGS\\"><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 24 24\\" height=\\"1em\\" width=\\"1em\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path fill=\\"none\\" stroke=\\"#000\\" stroke-width=\\"2\\" d=\\"M3.11111111,7.55555556 C4.66955145,4.26701301 8.0700311,2 12,2 C17.5228475,2 22,6.4771525 22,12 C22,17.5228475 17.5228475,22 12,22 L12,22 C6.4771525,22 2,17.5228475 2,12 M2,4 L2,8 L6,8 M9,16 L9,9 L7,9.53333333 M17,12 C17,10 15.9999999,8.5 14.5,8.5 C13.0000001,8.5 12,10 12,12 C12,14 13,15.5000001 14.5,15.5 C16,15.4999999 17,14 17,12 Z M14.5,8.5 C16.9253741,8.5 17,11 17,12 C17,13 17,15.5 14.5,15.5 C12,15.5 12,13 12,12 C12,11 12.059,8.5 14.5,8.5 Z\\"></path></svg></button><button class=\\"button__Clear-nas94j-5 hTVqGS\\"><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 24 24\\" height=\\"1em\\" width=\\"1em\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path fill=\\"none\\" stroke=\\"#000\\" stroke-width=\\"2\\" d=\\"M20.8888889,7.55555556 C19.3304485,4.26701301 15.9299689,2 12,2 C6.4771525,2 2,6.4771525 2,12 C2,17.5228475 6.4771525,22 12,22 L12,22 C17.5228475,22 22,17.5228475 22,12 M22,4 L22,8 L18,8 M9,16 L9,9 L7,9.53333333 M17,12 C17,10 15.9999999,8.5 14.5,8.5 C13.0000001,8.5 12,10 12,12 C12,14 13,15.5000001 14.5,15.5 C16,15.4999999 17,14 17,12 Z M14.5,8.5 C16.9253741,8.5 17,11 17,12 C17,13 17,15.5 14.5,15.5 C12,15.5 12,13 12,12 C12,11 12.059,8.5 14.5,8.5 Z\\"></path></svg></button><div class=\\"controls__ButtonContainer-sc-1r0nwd9-3 eDEhpx\\"><button class=\\"button__Clear-nas94j-5 eXrhoN\\"><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" version=\\"1.1\\" viewBox=\\"0 0 16 16\\" height=\\"1em\\" width=\\"1em\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M11.243 12.993c-0.192 0-0.384-0.073-0.53-0.22-0.293-0.293-0.293-0.768 0-1.061 2.047-2.047 2.047-5.378 0-7.425-0.293-0.293-0.293-0.768 0-1.061s0.768-0.293 1.061 0c1.275 1.275 1.977 2.97 1.977 4.773s-0.702 3.498-1.977 4.773c-0.146 0.146-0.338 0.22-0.53 0.22v0zM8.578 11.578c-0.192 0-0.384-0.073-0.53-0.22-0.293-0.293-0.293-0.768 0-1.061 1.267-1.267 1.267-3.329 0-4.596-0.293-0.293-0.293-0.768 0-1.061s0.768-0.293 1.061 0c1.852 1.852 1.852 4.865 0 6.718-0.146 0.146-0.338 0.22-0.53 0.22z\\"></path><path d=\\"M6.5 15c-0.13 0-0.258-0.051-0.354-0.146l-3.854-3.854h-1.793c-0.276 0-0.5-0.224-0.5-0.5v-5c0-0.276 0.224-0.5 0.5-0.5h1.793l3.854-3.854c0.143-0.143 0.358-0.186 0.545-0.108s0.309 0.26 0.309 0.462v13c0 0.202-0.122 0.385-0.309 0.462-0.062 0.026-0.127 0.038-0.191 0.038z\\"></path></svg></button><div class=\\"volume-slider__Main-sc-1al9r8e-0 eGLFr\\"><div class=\\"volume-slider__Volume-sc-1al9r8e-1 iFKvmY\\"><div class=\\"volume-slider__VolumeBarContainer-sc-1al9r8e-2 eTfkTW\\"><div class=\\"volume-slider__VolumeBar-sc-1al9r8e-3 jViDXh\\"><div class=\\"volume-slider__VolumeBarCurrent-sc-1al9r8e-4 dkBzgC\\"></div></div><div class=\\"volume-slider__VolumeBarIndicator-sc-1al9r8e-5 FhCdU\\"></div></div></div></div></div></div><div class=\\"controls__Middle-sc-1r0nwd9-4 bvxcPu\\"><h3 class=\\"controls__Title-sc-1r0nwd9-5 hAfgYq\\">Testtitle <span>Snull:Enull null</span></h3></div><div class=\\"controls__Right-sc-1r0nwd9-6 kHPNEo\\"><button class=\\"button__Clear-nas94j-5 hTVqGS\\"><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 24 24\\" height=\\"1em\\" width=\\"1em\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path fill=\\"none\\" stroke=\\"#000\\" stroke-width=\\"2\\" d=\\"M12,22 C17.5228475,22 22,17.5228475 22,12 C22,6.4771525 17.5228475,2 12,2 C6.4771525,2 2,6.4771525 2,12 C2,17.5228475 6.4771525,22 12,22 Z M12,15 L12,14 C12,13 12,12.5 13,12 C14,11.5 15,11 15,9.5 C15,8.5 14,7 12,7 C10,7 9,8.26413718 9,10 M12,16 L12,18\\"></path></svg><div class=\\"popup__Main-al6sl5-0 hIqMKL\\"><div class=\\"popup__Dialog-al6sl5-1 iSKUtS\\">Hi there. Something wrong?</div></div></button><button class=\\"button__Clear-nas94j-5 lfvVAq\\"><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 24 24\\" height=\\"1em\\" width=\\"1em\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path fill=\\"none\\" stroke=\\"#000\\" stroke-width=\\"2\\" d=\\"M1,2 L22,2 L22,18 L14,18 L6,22 L6,18 L1,18 L1,2 Z M6,10 L7,10 L7,11 L6,11 L6,10 Z M11,10 L12,10 L12,11 L11,11 L11,10 Z M16,10 L17,10 L17,11 L16,11 L16,10 Z\\"></path></svg></button><button class=\\"button__Clear-nas94j-5 lluGkg\\"><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 24 24\\" height=\\"1em\\" width=\\"1em\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path fill=\\"none\\" stroke=\\"#000\\" stroke-width=\\"2\\" d=\\"M10,14 L2,22 M1,15 L1,23 L9,23 M22,2 L14,10 M15,1 L23,1 L23,9\\"></path></svg></button></div></div></div>"`;

exports[`<ControlsContainer/> with no props & state should match the snapshot 1`] = `""`;
